js/src/ScatterBase.ts

```diff
@@ -663,8 +663,8 @@ export abstract class ScatterBase extends Mark {
         const yn = new y.constructor(y.length+1)
         xn.set(x)
         yn.set(y)
-        xn[x.length] = x_scale.scale.invert(curr_pos[0]);
-        yn[y.length] = y_scale.scale.invert(curr_pos[1]);
```

Comment:
I actually think this `scale` property should be private on scales. That would prevent this at least.